### PR TITLE
change most headers to be about module features

### PR DESCRIPTION
**Pull request by asymmetric**
Some headers could not be made about module features, and that's a strong signal that those sections should be removed, IMO.

### CODE REVIEW

File: source/tutorials/module-system/module-system.md [hunk 1]
```diff
@@ -833,7 +829,7 @@ The `lib.mkDefault` modifier sets the priority of its argument value to 1000, th
 This ensures that other values set for the same option will prevail.
 :::
 
-## Marker Styling: Color
+## The `either` and `enum` Types
```
Comment (fricklerhandwerk) [inline source/tutorials/module-system/module-system.md:832]:
```suggestion
## The `either` and `enum` types
```

Sentence-style capitalisation

**Review by fricklerhandwerk — APPROVED** [review]
Thanks for the cleanup. Let's get all of this out the door finally!

File: source/tutorials/module-system/module-system.md [hunk 2]
```diff
@@ -587,7 +585,7 @@ Make the following changes to `marker.nix`:
 +  };
 ```
 
-## Setting Option Values Within Other Modules
+## Defining Options in Other Modules
```
[the remark this answers is not in the document]
Comment (infinisil) [reply source/tutorials/module-system/module-system.md:588]: Yeah so
```suggestion
## Declaring Options in Other Modules
```

[the remark this answers is not in the document]
Comment (fricklerhandwerk) [reply source/tutorials/module-system/module-system.md:588]: No, we're not declaring options in this section.

**Pull request merged**